Core/Libraries/Source/Compression/CMakeLists.txt

```diff
@@ -34,7 +34,13 @@ target_link_libraries(core_compression PRIVATE
     corei_libraries_include
 )
 
+
 target_link_libraries(core_compression PUBLIC
     liblzhl
-    libzlib
 )
+
+find_package(ZLIB REQUIRED)
+if (ZLIB_FOUND)
+    target_link_libraries(core_compression PUBLIC ZLIB::ZLIB)
+    target_compile_definitions(core_compression PUBLIC RTS_HAS_ZLIB)
```

Comment:
for stlport we use "USING_STLPORT". I suggest to make it consistent one way or another.

Reply:
Ok. We can rename the USING_STLPORT in another change then.

Reply:
I see you renamed it in this change. I suggest we rename it later, because there are a multitude of changes touching related code right now.